sconstruct

```diff
@@ -1,22 +1,61 @@
 ###
-#This file is a part of the NVDA project.
-#URL: https://www.nvaccess.org/
-#Copyright 2010-2019 NV Access Limited, Babbage B.V.
-#This program is free software: you can redistribute it and/or modify
-#it under the terms of the GNU General Public License version 2.0, as published by
-#the Free Software Foundation.
-#This program is distributed in the hope that it will be useful,
-#but WITHOUT ANY WARRANTY; without even the implied warranty of
-#MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
-#This license can be found at:
-#http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# This file is a part of the NVDA project.
+# URL: https://www.nvaccess.org/
+# Copyright 2010-2020 NV Access Limited, Babbage B.V.
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2.0, as published by
+# the Free Software Foundation.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# This license can be found at:
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
 ###
 
 import sys
 import os
+import platform
+
+# Variables for storing required version of Python, and the version which is used to run this script.
+requiredPythonMajor ="3"
+requiredPythonMinor = "7"
+requiredPythonArchitecture = "32bit"
+installedPythonMajor = str(sys.version_info.major)
+installedPythonMinor = str(sys.version_info.minor)
+installedPythonArchitecture = platform.architecture()[0]
+# Ensure that we are running with required version of Python, otherwise inform the user and exit.
+if (
+	installedPythonArchitecture != requiredPythonArchitecture
+	or installedPythonMajor != requiredPythonMajor
+	or installedPythonMinor != requiredPythonMinor
+):
+	print(

```

Comment:
I understand why you used print here, it's just what the original script did. However, given what we do elsewhere in the scons environment, I think it makes sense to raise RuntimeError here.

Please:

1. As the message spans multiple lines, store it in its own variable
2. Then, raise a RuntimeError with that variable as message.

Using old style string formatting is fine of course.